bundles/org.openhab.binding.radiothermostat/src/main/java/org/openhab/binding/radiothermostat/internal/discovery/RadioThermostatDiscoveryService.java

```diff
@@ -0,0 +1,303 @@
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.radiothermostat.internal.discovery;
+
+import static org.eclipse.jetty.http.HttpMethod.GET;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.DatagramPacket;
+import java.net.Inet4Address;
+import java.net.InetAddress;
+import java.net.MalformedURLException;
+import java.net.MulticastSocket;
+import java.net.NetworkInterface;
+import java.net.SocketException;
+import java.net.SocketTimeoutException;
+import java.net.URL;
+import java.net.UnknownHostException;
+import java.util.Enumeration;
+import java.util.Map;
+import java.util.Scanner;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.smarthome.config.discovery.AbstractDiscoveryService;
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.openhab.binding.radiothermostat.internal.RadioThermostatBindingConstants;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * The {@link RadioThermostatDiscoveryService} is responsible for discovery of
+ * RadioThermostats on the local network
+ * 
+ * @author William Welliver - Initial contribution
+ * @author Dan Cunningham - Refactoring and Improvements
+ * @author Bill Forsyth - Modified for the RadioThermostat's peculiar discovery mode
+ * @author Michael Lobstein - Cleanup for RadioThermostat
+ * 
+ */
+
+@NonNullByDefault
+public class RadioThermostatDiscoveryService extends AbstractDiscoveryService {
+    private final Logger logger = LoggerFactory.getLogger(RadioThermostatDiscoveryService.class);
+    private static final String RADIOTHERMOSTAT_DISCOVERY_MESSAGE = "TYPE: WM-DISCOVER\r\nVERSION: 1.0\r\n\r\nservices:com.marvell.wm.system*\r\n\r\n";
+
+    private static final String SSDP_MATCH = "WM-NOTIFY";
+    private static final int BACKGROUND_SCAN_INTERVAL_SECONDS = 300;
+
+    private @Nullable ScheduledFuture<?> scheduledFuture = null;
+
+    private final HttpClient httpClient;
+
+    @Activate
+    public RadioThermostatDiscoveryService(@Reference HttpClient httpClient) {
+        super(RadioThermostatBindingConstants.SUPPORTED_THING_TYPES_UIDS, 30, true);
+        this.httpClient = httpClient;
```

Comment:
I forgot that you can't inject a HttpClient, only a HttpClientFactory. So my apologies.
```suggestion
    @Activate
    public RadioThermostatDiscoveryService(@Reference HttpClientFactory httpClientFactory) {
        super(RadioThermostatBindingConstants.SUPPORTED_THING_TYPES_UIDS, 30, true);
        this.httpClient = httpClientFactory.getCommonHttpClient();
```

Reply:
removed and reverted the factory class back to the original incarnation.